community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/symbols/TypeSpec.scala

```diff
@@ -36,6 +36,7 @@ object TypeSpec {
     CTNumber,
     CTPath,
     CTRelationship,
+    CTGeometry,
```

Comment:
With this code I'm experimenting with the idea that if I keep things somewhat general, and then specialize when appropriate, I will have a better understanding of the future compatibility options. I think that it is likely that most of the code in Spatial will work with the supertype 'Geometry' and only a few limited parts will need to work with specialized types. This is, to my mind, a separate concern than the externally facing parts like the point() function, which obviously creates a very specialized point instance. But once it enters the bowls of the database, I think we complicate things by needing to support many types, when one will do.